Userland/Games/Chess/ChessWidget.cpp

```diff
@@ -634,46 +634,49 @@ bool ChessWidget::export_pgn(StringView export_path) const
     }
     auto& file = *file_or_error.value();
 
+    StringBuilder builder;
     // Tag Pair Section
-    file.write("[Event \"Casual Game\"]\n");
-    file.write("[Site \"SerenityOS Chess\"]\n");
-    file.write(String::formatted("[Date \"{}\"]\n", Core::DateTime::now().to_string("%Y.%m.%d")));
-    file.write("[Round \"1\"]\n");
+    builder.append("[Event \"Casual Game\"]\n");
+    builder.append("[Event \"Casual Game\"]\n");
+    builder.append("[Site \"SerenityOS Chess\"]\n");
+    builder.append(String::formatted("[Date \"{}\"]\n", Core::DateTime::now().to_string("%Y.%m.%d")));
+    builder.append("[Round \"1\"]\n");
 
     String username(getlogin());
     const String player1 = (!username.is_empty() ? username : "?");
     const String player2 = (!m_engine.is_null() ? "SerenityOS ChessEngine" : "?");
-    file.write(String::formatted("[White \"{}\"]\n", m_side == Chess::Color::White ? player1 : player2));
-    file.write(String::formatted("[Black \"{}\"]\n", m_side == Chess::Color::Black ? player1 : player2));
+    builder.append(String::formatted("[White \"{}\"]\n", m_side == Chess::Color::White ? player1 : player2));
+    builder.append(String::formatted("[Black \"{}\"]\n", m_side == Chess::Color::Black ? player1 : player2));
 
-    file.write(String::formatted("[Result \"{}\"]\n", Chess::Board::result_to_points(m_board.game_result(), m_board.turn())));
-    file.write("[WhiteElo \"?\"]\n");
-    file.write("[BlackElo \"?\"]\n");
-    file.write("[Variant \"Standard\"]\n");
-    file.write("[TimeControl \"-\"]\n");
-    file.write("[Annotator \"SerenityOS Chess\"]\n");
-    file.write("\n");
+    builder.append(String::formatted("[Result \"{}\"]\n", Chess::Board::result_to_points(m_board.game_result(), m_board.turn())));
```

Comment:
For future reference: We also have `StringBuilder::appendff`, which saves an allocation. I don't think it matters here, but now you know :D

Reply:
Oh that's pretty neat! I went ahead and implemented it anyway - it looks a little nicer :^)